diff -ru go.orig/src/crypto/x509/root_unix.go go/src/crypto/x509/root_unix.go
--- go.orig/src/crypto/x509/root_unix.go	2015-09-23 04:37:36.000000000 +0000
+++ go/src/crypto/x509/root_unix.go	2016-02-24 16:45:18.081961839 +0000
@@ -17,6 +17,7 @@
 	"/usr/local/share/certs/ca-root-nss.crt", // FreeBSD/DragonFly
 	"/etc/pki/tls/cacert.pem",                // OpenELEC
 	"/etc/certs/ca-certificates.crt",         // Solaris 11.2+
+	"@cacerts@", // Bldr Package
 }
 
 // Possible directories with certificate files; stop after successfully
