diff -ru glibc-2.22-orig/elf/ldconfig.c glibc-2.22/elf/ldconfig.c
--- glibc-2.22-orig/elf/ldconfig.c	2015-12-12 02:05:44.630174539 +0000
+++ glibc-2.22/elf/ldconfig.c	2015-12-12 02:42:45.537441848 +0000
@@ -51,7 +51,7 @@
 #endif
 
 #ifndef LD_SO_CONF
-# define LD_SO_CONF SYSCONFDIR "/ld.so.conf"
+# define LD_SO_CONF PREFIX "/etc/ld.so.conf"
 #endif
 
 /* Get libc version number.  */
diff -ru glibc-2.22-orig/elf/Makefile glibc-2.22/elf/Makefile
--- glibc-2.22-orig/elf/Makefile	2015-12-12 02:05:44.626174409 +0000
+++ glibc-2.22/elf/Makefile	2015-12-12 02:44:17.044734055 +0000
@@ -443,13 +443,13 @@
 
 $(objpfx)ldconfig: $(ldconfig-modules:%=$(objpfx)%.o)
 
-SYSCONF-FLAGS := -D'SYSCONFDIR="$(sysconfdir)"'
-CFLAGS-ldconfig.c = $(SYSCONF-FLAGS) -D'LIBDIR="$(libdir)"' \
+PREFIX-FLAGS := -D'PREFIX="$(prefix)"'
+CFLAGS-ldconfig.c = $(PREFIX-FLAGS) -D'LIBDIR="$(libdir)"' \
 		    -D'SLIBDIR="$(slibdir)"'
 libof-ldconfig = ldconfig
-CFLAGS-dl-cache.c = $(SYSCONF-FLAGS)
-CFLAGS-cache.c = $(SYSCONF-FLAGS)
-CFLAGS-rtld.c = $(SYSCONF-FLAGS)
+CFLAGS-dl-cache.c = $(PREFIX-FLAGS)
+CFLAGS-cache.c = $(PREFIX-FLAGS)
+CFLAGS-rtld.c = $(PREFIX-FLAGS)
 
 cpp-srcs-left := $(all-rtld-routines:=.os)
 lib := rtld
diff -ru glibc-2.22-orig/elf/rtld.c glibc-2.22/elf/rtld.c
--- glibc-2.22-orig/elf/rtld.c	2015-12-12 02:05:44.630174539 +0000
+++ glibc-2.22/elf/rtld.c	2015-12-12 02:48:31.673870395 +0000
@@ -1513,7 +1513,7 @@
      open().  So we do this first.  If it succeeds we do almost twice
      the work but this does not matter, since it is not for production
      use.  */
-  static const char preload_file[] = "/etc/ld.so.preload";
+  static const char preload_file[] = "@prefix@/etc/ld.so.preload";
   if (__glibc_unlikely (__access (preload_file, R_OK) == 0))
     {
       /* Read the contents of the file.  */
diff -ru glibc-2.22-orig/sysdeps/generic/dl-cache.h glibc-2.22/sysdeps/generic/dl-cache.h
--- glibc-2.22-orig/sysdeps/generic/dl-cache.h	2015-12-12 02:05:44.846181518 +0000
+++ glibc-2.22/sysdeps/generic/dl-cache.h	2015-12-12 02:45:08.982600296 +0000
@@ -28,7 +28,7 @@
 #endif
 
 #ifndef LD_SO_CACHE
-# define LD_SO_CACHE SYSCONFDIR "/ld.so.cache"
+# define LD_SO_CACHE PREFIX "/etc/ld.so.cache"
 #endif
 
 #ifndef add_system_dir
