diff -ru glibc-2.22.orig/stdio-common/Makefile glibc-2.22/stdio-common/Makefile
--- glibc-2.22.orig/stdio-common/Makefile	2015-08-05 06:42:21.000000000 +0000
+++ glibc-2.22/stdio-common/Makefile	2015-12-17 04:51:00.303026327 +0000
@@ -107,9 +107,9 @@
 # GNU extension.  The latter are needed, though, when internal headers
 # are used.  So made sure we see the installed headers first.
 CFLAGS-scanf15.c = -I../libio -I../stdlib -I../wcsmbs -I../time -I../string \
-		   -I../wctype
+		   -I../wctype -I$(common-objpfx)misc
 CFLAGS-scanf17.c = -I../libio -I../stdlib -I../wcsmbs -I../time -I../string \
-		   -I../wctype
+		   -I../wctype -I$(common-objpfx)misc
 
 CPPFLAGS += $(libio-mtsafe)
 
