diff -ru gcc-5.2.0.orig/gcc/cppdefault.c gcc-5.2.0/gcc/cppdefault.c
--- gcc-5.2.0.orig/gcc/cppdefault.c	2015-01-05 12:33:28.000000000 +0000
+++ gcc-5.2.0/gcc/cppdefault.c	2015-12-23 20:18:54.631026327 +0000
@@ -35,6 +35,8 @@
 # undef CROSS_INCLUDE_DIR
 #endif
 
+#undef LOCAL_INCLUDE_DIR
+
 const struct default_include cpp_include_defaults[]
 #ifdef INCLUDE_DEFAULTS
 = INCLUDE_DEFAULTS;
diff -ru gcc-5.2.0.orig/gcc/gcc.c gcc-5.2.0/gcc/gcc.c
--- gcc-5.2.0.orig/gcc/gcc.c	2015-03-10 09:37:41.000000000 +0000
+++ gcc-5.2.0/gcc/gcc.c	2015-12-23 20:20:08.027026327 +0000
@@ -1261,10 +1261,10 @@
 /* Default prefixes to attach to command names.  */
 
 #ifndef STANDARD_STARTFILE_PREFIX_1
-#define STANDARD_STARTFILE_PREFIX_1 "/lib/"
+#define STANDARD_STARTFILE_PREFIX_1 ""
 #endif
 #ifndef STANDARD_STARTFILE_PREFIX_2
-#define STANDARD_STARTFILE_PREFIX_2 "/usr/lib/"
+#define STANDARD_STARTFILE_PREFIX_2 ""
 #endif
 
 #ifdef CROSS_DIRECTORY_STRUCTURE  /* Don't use these prefixes for a cross compiler.  */
